File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/RoutineAssignmentRepository.swift
//
//  RoutineAssignmentRepository.swift
//  RoutineChart
//
//  Repository protocol for RoutineAssignment data access
//

import Foundation

protocol RoutineAssignmentRepository {
    func create(_ assignment: RoutineAssignment) async throws
    func get(id: String) async throws -> RoutineAssignment?
    func update(_ assignment: RoutineAssignment) async throws
    func getAll(familyId: String) async throws -> [RoutineAssignment]
    func getByChild(familyId: String, childId: String) async throws -> [RoutineAssignment]
    func getByRoutine(familyId: String, routineId: String) async throws -> [RoutineAssignment]
    func softDelete(id: String) async throws
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/RoutineAssignmentRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.RoutineAssignment
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for RoutineAssignment operations
 */
interface RoutineAssignmentRepository {
    suspend fun create(assignment: RoutineAssignment)

    suspend fun update(assignment: RoutineAssignment)

    suspend fun getById(id: String): RoutineAssignment?

    suspend fun getByFamilyId(familyId: String): List<RoutineAssignment>

    suspend fun getActiveByChildId(childId: String): List<RoutineAssignment>

    fun observeActiveByChildId(childId: String): Flow<List<RoutineAssignment>>

    suspend fun getByRoutineId(routineId: String): List<RoutineAssignment>
}
